src/clr-angular/data/tree-view/tree-node.ts

```diff
@@ -153,6 +155,10 @@ export class ClrTreeNode extends AbstractTreeSelection implements OnDestroy {
     return this.nodeExpand.expanded ? 'down' : 'right';
   }
 
+  public get caretTitle(): string {
+    return this.nodeExpand.expanded ? this.commonStrings.collapse : this.commonStrings.expand;
```

Comment:
We've got this logic in the template in most other places. I'd like it to be consistent either way, though it is nice to have it in the controller for testability.

Reply:
I decided to not refactor anything to keep the PR focused. You'll notice the caret class `this.nodeExpand.expanded ? 'down' : 'right';` is in the template often, and in the controller sometimes. To minimize the footprint of this PR, I went with the convention in place *for that component* whenever I added the caret title.

I'm happy to open an issue to unify all this in a future PR, if you feel it's a clean-up we need to do.

Reply:
That sounds fair, and we can decide to clean up later. I hadn't really thought about this before, might be something for convention guidelines in the future.